lib/hanami/cli/commands/app/generate/migration.rb

```diff
@@ -9,16 +9,36 @@ module Generate
           # @api private
           class Migration < Command
             argument :name, required: true, desc: "Migration name"
+            option :gateway, desc: "Generate migration for gateway"
 
             example [
               %(create_posts),
               %(add_published_at_to_posts),
               %(create_users --slice=admin),
+              %(create_comments --slice=admin --gateway=extra),
             ]
 
             def generator_class
               Generators::App::Migration
             end
+
+            def call(name:, slice: nil, gateway: nil)
+              if slice
+                generator.call(
+                  key: name,
+                  namespace: slice,
+                  base_path: fs.join("slices", inflector.underscore(slice)),
+                  gateway: gateway
+                )
+              else
+                generator.call(
+                  key: name,
+                  namespace: app.namespace,
+                  base_path: "app",
+                  gateway: gateway
+                )
+              end
+            end
```

Comment:
@cllns a note for you — to be able to pass this extra `gateway: gateway` arg when calling the generator, I had to copy the entire `#call` implementation from the base generate command class. It would be good if we could allow individual generate command classes to be able to provide extra args to the generator without having to do this. An opportunity for a refactor in the future?